internal/k8s/exec.go

```diff
@@ -34,6 +35,9 @@ type ExecParameters struct {
 	Pod       string
 	Container string
 	Command   []string
+	TTY       bool
+	Stdin     io.Reader
+	Stdout    io.Writer
 }
 
 func (c *Client) execInPod(ctx context.Context, p ExecParameters) (*ExecResult, error) {
```

Comment:
`ctx` is unused here, so this would be an appropriate place to inject `CtrlCReader`. As suggested above, YAGNI on this function taking TTY/Stdin/Stdout from the caller. Just provide cancelation out of the box, will drive the complexity down a considerable amount, and should also minimize the amount of code churn in this PR.

Reply:
this was needed for the monitor / screen scraping, will refactor.

Reply:
Done.